core/auth/index.ts

```diff
@@ -148,6 +154,10 @@ async function loginWithJwt(credentials: unknown): Promise<User | null> {
   };
 }
 
+function loginWithSessionSync(credentials: unknown): User {
+  return SessionSyncCredentials.parse(credentials);
+}
```

Comment:
This is a little awkward... Usually you should handle the validation of the sign in here but because we are going to move in a direction where there will alway be a session, this `CredentialsProvider` will actually be removed and we will just update the session accordingly. We also need to handle things like the redirect and any API validation errors so it made more sense to do the validation in the route handler, rather than in the `signIn` function.